crates/holochain_integrity_types/src/capability/grant.rs

```diff
@@ -111,13 +111,19 @@ impl CapGrant {
                     // The grant has no assignees so is always valid…
                     _ => true,
                 }
-                // The secret needs to match…
-                && match access {
-                    // Unless the extern is unrestricted.
-                    CapAccess::Unrestricted => true,
-                    // note the PartialEq implementation is constant time for secrets
-                    CapAccess::Transferable { secret, .. } => check_secret.map(|given| secret == given).unwrap_or(false),
-                    CapAccess::Assigned { secret, .. } => check_secret.map(|given| secret == given).unwrap_or(false),
+                // The secret needs to match, if provided
+                && {
+                    if let Some(given_secret) = check_secret {
+                        match access {
+                            // Unless the extern is unrestricted.
+                            CapAccess::Unrestricted => true,
+                            // note the PartialEq implementation is constant time for secrets
+                            CapAccess::Transferable { secret, .. } => secret == given_secret,
```

Comment:
Transferable requires a secret, the check must not be skipped by not providing one.